File: Csharp/ObjectiveTA/ObjectiveTA/Objects/Input/PriceSource.cs
using System;
namespace ObjectiveTA.Objects.Input
{
    public enum PriceSource{Open, High, Low, Close}

    static class PriceSourceExtensions
    {
        /// <summary>
        /// Gets the selected price from a candle stick.
        /// </summary>
        /// <returns>The value from candle stick.</returns>
        /// <param name="p">P.</param>
        /// <param name="c">C.</param>
        public static double GetValueFromCandleStick(this PriceSource p, CandleStick c)
        {
            switch(p)
            {
                case PriceSource.Open:
                    return c.Open;
                case PriceSource.High:
                    return c.High;
                case PriceSource.Low:
                    return c.Low;
                case PriceSource.Close:
                    return c.Close;
                default:
                    Exception exception = new Exception("Invalid price source value");
                    throw(exception);
            }
        }

        public static double[] GetArrayFromCandleStickCollection(this PriceSource p, 
                                                                 CandleStickCollection c)
        {
            int count = c.Count;
            double[] priceArray = new double[count];

            switch (p)
            {
                case PriceSource.Open:
                    for (int i = 0; i < count - 1; i++){
                        priceArray[i] = c[i].Open;
                    }
                    return priceArray;
                case PriceSource.High:
                    for (int i = 0; i < count - 1; i++)
                    {
                        priceArray[i] = c[i].High;
                    }
                    return priceArray;
                case PriceSource.Low:
                    for (int i = 0; i < count - 1; i++)
                    {
                        priceArray[i] = c[i].Low;
                    }
                    return priceArray;
                case PriceSource.Close:
                    for (int i = 0; i < count - 1; i++)
                    {
                        priceArray[i] = c[i].Close;
                    }
                    return priceArray;
                default:
                    Exception exception = new Exception("Invalid price source value");
                    throw (exception);
            }
        }

    }
}

File: cpp/Objects/Input/PriceSource.cpp
/*
 * PriceSource.cpp
 *
 *  Created on: 28 Aug 2018
 *      Author: will
 */

#include "PriceSource.h"

namespace objectiveTA::objects::input {


PriceSource PriceSource::Open() {
	this->priceSourceEnum::_open;
	return this;
}

PriceSource PriceSource::High() {
	this->priceSourceEnum::_high;
	return this;
}

PriceSource PriceSource::Low() {
	this->priceSourceEnum::_low;
	return this;
}

PriceSource PriceSource::Close() {
	this->priceSourceEnum::_close;
	return this;
}

vector<double> PriceSource::GetArrayFromCandleStickCollection(CandleStickCollection cs) {
	vector<double> priceArray = new vector<double>;

	switch(this->priceSourceEnum)
	{
		case priceSourceEnum::_open:
			break;
		case priceSourceEnum::_high:
			break;
		case priceSourceEnum::_low:
			break;
		case priceSourceEnum::_close:
			break;
	}

	for(int i = 0; i < static_cast<int>(cs.size()); i++) {

	}
	return priceArray;
}

} /* namespace objectiveTA */
